packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx

```diff
@@ -50,171 +53,233 @@ export interface IGatsbyImageData {
   placeholder?: Pick<PlaceholderProps, "sources" | "fallback">
 }
 
-let hasShownWarning = false
-
-export const GatsbyImageHydrator: FunctionComponent<GatsbyImageProps> = function GatsbyImageHydrator({
-  as: Type = `div`,
-  style,
-  className,
-  class: preactClass,
-  onStartLoad,
-  image,
-  onLoad: customOnLoad,
-  backgroundColor,
-  loading = `lazy`,
-  ...props
-}) {
-  if (!image) {
-    if (process.env.NODE_ENV === `development`) {
-      console.warn(`[gatsby-plugin-image] Missing image prop`)
+class GatsbyImageHydrator extends Component<
+  GatsbyImageProps,
+  { isLoading: boolean; isLoaded: boolean }
+> {
+  root: RefObject<HTMLImageElement | undefined> = createRef<
+    HTMLImageElement | undefined
+  >()
+  hydrated: MutableRefObject<boolean> = { current: false }
+  lazyHydrator: () => void | null = null
+  ref = createRef<HTMLImageElement>()
+  unobserveRef: (element: RefObject<HTMLElement | undefined>) => Unobserver
```

Comment:
I think this is the wrong type. The return value from `intersectionObserver()` is `Unobserver`